commons/pkg/auditlogger/auditlogger.go

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package auditlogger provides file-based audit logging for write operations in NVSentinel components.
+// It automatically captures and logs HTTP write operations (POST, PUT, PATCH, DELETE) with timestamps,
+// component identification, request details, and response codes. The package supports automatic log
+// rotation using lumberjack and provides configurable options through environment variables for log
+// directory, file size limits, retention policies, and optional request body logging.
+package auditlogger
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"gopkg.in/natefinch/lumberjack.v2"
+)
+
+const (
+	DefaultAuditLogDir     = "/var/log/nvsentinel"
+	EnvAuditLogBasePath    = "AUDIT_LOG_BASE_PATH"
+	EnvAuditEnabled        = "AUDIT_ENABLED"
+	EnvAuditLogRequestBody = "AUDIT_LOG_REQUEST_BODY"
+	EnvPodName             = "POD_NAME"
+	EnvAuditLogMaxSize     = "AUDIT_LOG_MAX_SIZE_MB"
+	EnvAuditLogMaxBackups  = "AUDIT_LOG_MAX_BACKUPS"
+	EnvAuditLogMaxAge      = "AUDIT_LOG_MAX_AGE_DAYS"
+	EnvAuditLogCompress    = "AUDIT_LOG_COMPRESS"
+)
+
+// AuditEntry represents a single audit log entry for an HTTP write operation, capturing the
+// Timestamp (RFC3339 format), Component name, HTTP Method and URL, ResponseCode from the server,
+// and optionally the RequestBody if configured via AUDIT_LOG_REQUEST_BODY environment variable.
+type AuditEntry struct {
+	Timestamp    string `json:"timestamp"`
+	Component    string `json:"component"`
+	Method       string `json:"method"`
+	URL          string `json:"url"`
+	ResponseCode int    `json:"response_code,omitempty"`
+	RequestBody  string `json:"requestBody,omitempty"`
+}
+
+var (
+	auditLogger *lumberjack.Logger
+	component   string
+)
+
+func getEnvInt(key string, defaultVal int) int {
+	if v := os.Getenv(key); v != "" {
+		if i, err := strconv.Atoi(v); err == nil {
+			return i
+		}
+	}
+
+	return defaultVal
+}
+
+func getEnvBool(key string, defaultVal bool) bool {
```

Comment:
This function seems to be generic and can be used with any feature; can we refactor this into a utils folder?